packages/infra/lib/ihe-stack.ts

```diff
@@ -5,20 +5,27 @@ import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
 import * as ec2 from "aws-cdk-lib/aws-ec2";
 import * as r53 from "aws-cdk-lib/aws-route53";
 import * as r53_targets from "aws-cdk-lib/aws-route53-targets";
+import * as sns from "aws-cdk-lib/aws-sns";
 import { Construct } from "constructs";
 import { EnvConfig } from "../config/env-config";
 import { createLambda } from "./shared/lambda";
-import { setupLambdasLayers, LambdaLayers } from "./shared/lambda-layers";
+import { LambdaLayers, setupLambdasLayers } from "./shared/lambda-layers";
 
 interface IHEStackProps extends StackProps {
   config: EnvConfig;
-  vpc: ec2.IVpc;
-  alarmAction: SnsAction | undefined;
+  version: string | undefined;
 }
 
 export class IHEStack extends Stack {
   constructor(scope: Construct, id: string, props: IHEStackProps) {
     super(scope, id, props);
+
+    const vpcId = props.config.iheGateway?.vpcId;
+    if (!vpcId) throw new Error("Missing VPC ID for IHE stack");
+    const vpc = ec2.Vpc.fromLookup(this, "APIVpc", { vpcId });
```

Comment:
OOC, why is this approach preferred to passing in the VPC in the IHEstack params?

Reply:
It created a dependency between the stacks, we had issues trying to removing/moving stuff.